design/backend/templates/views/payments/components/cc_processors/razorpay/razorpay.tpl

```diff
@@ -49,16 +56,45 @@
     </script>
 <div>
 
-<div class="form-field">
+<div class="form-field" style="display:none">
     <label for="webhook_secret">{__("rzp_webhook_secret")}:</label>
     <input type="text" name="payment_data[processor_params][webhook_secret]" id="webhook_secret" value="{$processor_params.webhook_secret}" class="input-text" />
     <div style="font-weight: bold; font-style: italic;">This field has to match with the same secret, set in <a href="https://dashboard.razorpay.com/#/app/webhooks">https://dashboard.razorpay.com/#/app/webhooks</a></div>
 </div>
 
-<div class="form-field">
-    <label for="iframe_mode_{$payment_id}">{__("iframe_mode")}:</label>
-    <select name="payment_data[processor_params][iframe_mode]" id="iframe_mode_{$payment_id}">
-        <option value="Y" {if $processor_params.iframe_mode == "Y"}selected="selected"{/if}>{__("enabled")}</option>
-        <option value="N" {if $processor_params.iframe_mode == "N"}selected="selected"{/if}>{__("disabled")}</option>
-    </select>
+<div class="form-field" style="display:none">
+    <label for="webhook_flag">:</label>
+    <input type="text" name="payment_data[processor_params][webhook_flag]" id="webhook_flag" value="{$processor_params.webhook_flag}" class="input-text" />
 </div>
+<script type='text/javascript'>
+
+$('input[type="submit"]').click(function( event ) {
+   
+   $dt = Math.floor(Date.now() / 1000);
+   $('#webhook_flag').val($dt);
+   $secret = Math.random().toString(36).slice(2, 7);
+   $('#webhook_secret').val($secret);
+   $keyid = $('#key_id').val();
+   $keysecret = $('#key_secret').val();
+   $webhookurl  = $('#webhook_url').val();
+    
+   $.ajax({
+       url: "http://localhost/cscart/admin.php?dispatch=razorpay.manage", 
+       type: 'GET',
+        data: { 
+            keyid: $keyid, 
+            keysecret : $keysecret,
```

Comment:
@jampulanaveen , why key secret in ajax?

Reply:
in controller we do not have access to the $processor_data, so i am sending through ajax

Reply:
@jampulanaveen , if it's not there please read it from DB. we should not expose the API secret anywhere.

Reply:
@ramth05 made the changes, now fetching key_id and key_secret from db